internal/bminventory/inventory.go

```diff
@@ -833,13 +888,21 @@ func (b *bareMetalInventory) DeregisterClusterInternal(ctx context.Context, para
 		return common.NewApiError(http.StatusNotFound, err)
 	}
 
+	// AMS subscription is created only for day1 clusters
+	if *cluster.Kind == models.ClusterKindCluster {
+		if err := b.deleteReservedAMSSubscription(ctx, cluster.AmsSubscriptionID); err != nil {
+			log.WithError(err).Errorf("Failed to delete AMS subscription %s for non-active cluster %s", cluster.AmsSubscriptionID, params.ClusterID)
+			return common.NewApiError(http.StatusInternalServerError, err)
```

Comment:
just a thought, maybe it should be a warning instead of an error? I mean, is it crucial that the subscription gets deleted when removing a cluster? isn't it something that gets cleaned up by AMS later on any way?

Reply:
It is not deleted by AMS, this is the reason this code was added.
AMS will only archive `active` subscriptions, we update subscriptions to `active` only on CompleteInstallation API, therefore, on each cluster deregistration we need to make sure that we clean the `reserved` subscriptions because no one else will clean them.
If the installation succeed, then we will update the subscription to `active` status and AMS's reconcile will do the rest.